tests/backup/backup_helper.go

```diff
@@ -4171,3 +4174,179 @@ func CreateRuleForBackupWithMultipleApplications(orgID string, appList []string,
+		Uid:           backupUid,
+	}
+	_, err := backupDriver.DeleteBackupSchedule(ctx, bkpScheduleDeleteRequest)
+	if err != nil {
+		return err
+	}
+	clusterReq := &api.ClusterInspectRequest{OrgId: orgID, Name: clusterName, IncludeSecrets: true, Uid: clusterUid}
+	clusterResp, err := backupDriver.InspectCluster(ctx, clusterReq)
+	if err != nil {
+		return err
+	}
+	clusterObj := clusterResp.GetCluster()
+	namespace := "*"
+	err = backupDriver.WaitForBackupScheduleDeletion(ctx, backupScheduleName, namespace, orgID,
+		clusterObj,
+		backupDeleteTimeout,
+		RetrySeconds*time.Second)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetAllBackupsOfUsersFromAdmin get all backups of given ownerID from admin context.
+func GetAllBackupsOfUsersFromAdmin(ownerID []string, adminCtx context.Context) ([]string, error) {
+	backupNames := make([]string, 0)
+	backupDriver := Inst().Backup
+
+	backupEnumerateReq := &api.BackupEnumerateRequest{
+		OrgId: orgID,
+		EnumerateOptions: &api.EnumerateOptions{
+			Owners: ownerID,
+		},
+	}
+	currentBackups, err := backupDriver.EnumerateBackup(adminCtx, backupEnumerateReq)
+	if err != nil {
+		return nil, err
+	}
+	for _, backup := range currentBackups.GetBackups() {
+		backupNames = append(backupNames, backup.GetName())
+	}
+	return backupNames, nil
+}
+
+// DeleteCluster deletes/de-registers cluster from px-backup
+func DeleteClusterWithUid(clusterName string, clusterUid string, orgID string, ctx context.Context, cleanupBackupsRestores bool) error {
+
+	backupDriver := Inst().Backup
+	clusterDeleteReq := &api.ClusterDeleteRequest{
+		OrgId:          orgID,
+		Name:           clusterName,
+		DeleteBackups:  cleanupBackupsRestores,
+		DeleteRestores: cleanupBackupsRestores,
+		Uid:            clusterUid,
+	}
+	_, err := backupDriver.DeleteCluster(ctx, clusterDeleteReq)
+	if err != nil {
+		return err
+	}
+	err = backupDriver.WaitForClusterDeletion(ctx, clusterName, clusterUid, orgID, clusterDeleteTimeout, clusterCreationRetryTime)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateInvalidCloudCredential creates cloud credentials with invalid paramaters
+func createInvalidCloudCredential(credName string, uid, orgID string, ctx context.Context, kubeconfig ...string) error {
+	log.Infof("Create cloud credential with name [%s] for org [%s] ", credName, orgID)
+	var credCreateRequest *api.CloudCredentialCreateRequest
+	credCreateRequest = &api.CloudCredentialCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  credName,
+			Uid:   uid,
+			OrgId: orgID,
+		},
+		CloudCredential: &api.CloudCredentialInfo{
+			Type: api.CloudCredentialInfo_AWS,
+			Config: &api.CloudCredentialInfo_AwsConfig{
+				AwsConfig: &api.AWSConfig{
+					AccessKey: "admin",
+					SecretKey: backup.PxCentralAdminPwd + RandomString(10),
+				},
+			},
+		},
+	}
+	_, err := Inst().Backup.CreateCloudCredential(ctx, credCreateRequest)
+	if err != nil {
+		if strings.Contains(err.Error(), "already exists") {
+			return nil
```

Comment:
How about?

err := fmt.Errorf("failed to create invalid cloud credential with name [%s] in org [%s] with [AWS/S3] as provider", credName, orgID)

return err instead of nil?